### PR TITLE
add named port support

### LINKED ISSUE #2206: Network policy E2E fails
# Expected Behavior

# Actual Behavior

```txt
2022-12-30T09:53:48.0127828Z [38;5;9m[1mSummarizing 6 Failures:[0m
2022-12-30T09:53:48.0128460Z   [38;5;9m[FAIL][0m [0m[sig-network] Netpol [38;5;243mNetworkPolicy between server and client [0m[38;5;9m[1m[It] should allow ingress access from namespace on one named port [Feature:NetworkPolicy][0m[0m
2022-12-30T09:53:48.0128842Z   [38;5;243m/home/runner/go/pkg/mod/k8s.io/kubernetes@v1.26.0/test/e2e/network/netpol/test_helper.go:126[0m
2022-12-30T09:53:48.0129486Z   [38;5;9m[FAIL][0m [0m[sig-network] Netpol [38;5;243mNetworkPolicy between server and client [0m[38;5;9m[1m[It] should enforce except clause while egress access to server in CIDR block [Feature:NetworkPolicy][0m[0m
2022-12-30T09:53:48.0129860Z   [38;5;243m/home/runner/go/pkg/mod/k8s.io/kubernetes@v1.26.0/test/e2e/network/netpol/test_helper.go:126[0m
2022-12-30T09:53:48.0130500Z   [38;5;9m[FAIL][0m [0m[sig-network] Netpol [38;5;243mNetworkPolicy between server and client [0m[38;5;9m[1m[It] should allow ingress access on one named port [Feature:NetworkPolicy][0m[0m
2022-12-30T09:53:48.0130871Z   [38;5;243m/home/runner/go/pkg/mod/k8s.io/kubernetes@v1.26.0/test/e2e/network/netpol/test_helper.go:126[0m
2022-12-30T09:53:48.0131600Z   [38;5;9m[FAIL][0m [0m[sig-network] Netpol [38;5;243mNetworkPolicy between server and client [0m[38;5;9m[1m[It] should ensure an IP overlapping both IPBlock.CIDR and IPBlock.Except is allowed [Feature:NetworkPolicy][0m[0m
2022-12-30T09:53:48.0131961Z   [38;5;243m/home/runner/go/pkg/mod/k8s.io/kubernetes@v1.26.0/test/e2e/network/netpol/test_helper.go:126[0m
2022-12-30T09:53:48.0132532Z   [38;5;9m[FAIL][0m [0m[sig-network] Netpol [38;5;243mNetworkPolicy between server and client [0m[38;5;9m[1m[It] should allow egress access to server in CIDR block [Feature:NetworkPolicy][0m[0m
2022-12-30T09:53:48.0132900Z   [38;5;243m/home/runner/go/pkg/mod/k8s.io/kubernetes@v1.26.0/test/e2e/network/netpol/test_helper.go:126[0m
2022-12-30T09:53:48.0133714Z   [38;5;9m[FAIL][0m [0m[sig-network] Netpol [38;5;243mNetworkPolicy between server and client [0m[38;5;9m[1m[It] should allow egress access on one named port [Feature:NetworkPolicy][0m[0m
2022-12-30T09:53:48.0134091Z   [38;5;243m/home/runner/go/pkg/mod/k8s.io/kubernetes@v1.26.0/test/e2e/network/netpol/test_helper.go:126[0m
```
# Steps to Reproduce the Problem

1.
2.
3.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  (paste your output here)
  ```

- kube-ovn version:
  
  ```
  (paste your output here)
  ```

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
  **Output of `uname -r`:**
  ```
  (paste your output here)
  ```

### PR DESCRIPTION
**Pull request by changluyi**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

#### What type of this PR
Examples of user facing changes:
- Features
- Bug fixes
- Docs
- Tests

#### Which issue(s) this PR fixes:
Fixes #2206

### CODE REVIEW

File: pkg/controller/network_policy.go [hunk 1]
```diff
@@ -581,6 +587,37 @@ func (c *Controller) handleDeleteNp(key string) error {
...
+func (c *Controller) fetchNamedPorts(namespace string) (map[string]int32, error) {
+	namedPortMap := map[string]int32{}
+
+	pods, err := c.podsLister.Pods(namespace).List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("failed to list pods, %v", err)
+	}
+
+	for _, pod := range pods {
+		if !isPodAlive(pod) || pod.Spec.HostNetwork {
```
Comment (oilbeater) [inline pkg/controller/network_policy.go:599]: Not need to skip hostnetwork pods

File: pkg/controller/network_policy.go [hunk 2]
```diff
@@ -196,6 +196,12 @@ func (c *Controller) handleUpdateNp(key string) error {
 		return err
 	}
 
+	namedPortMap, err := c.fetchNamedPorts(np.Namespace)
```
Comment (oilbeater) [inline pkg/controller/network_policy.go:199]: We should move it to pod handler to reduce duplicated computation

**Review by oilbeater — CHANGES_REQUESTED** [review]

**Comment by oilbeater** [comment]
Cyclonus Network Policy E2E (ipv6) e2e failed

File: pkg/controller/network_policy.go (hunk 2 above)
Comment (changluyi) [reply pkg/controller/network_policy.go]: I refactored it

**Comment by zhangzujian** [comment]
https://github.com/kubeovn/kube-ovn/blob/master/.github/workflows/build-x86-image.yaml#L1069 should be uncommented.

**Review by oilbeater — APPROVED** [review]

**Pull request merged**